internal/dag/secret.go

```diff
@@ -82,3 +83,21 @@ func decodePEM(in []byte) (*pem.Block, error) {
 	}
 	return block, nil
 }
+
+func decodePEMPrivateKey(in []byte) (*pem.Block, error) {
+	var found bool
+	var block *pem.Block
+	for len(in) > 0 {
+		if block, in = pem.Decode(in); block == nil {
+			return nil, errors.New("failed to parse PEM")
+		}
+		if strings.HasSuffix(block.Type, "PRIVATE KEY") {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil, errors.New("failed to locate private key")
+	}
+	return block, nil
```

Comment:
there is an opportunity to be ""smarter" with the outbound block and switch on type to identify the parsing method.